### PR TITLE
Relationships

### PR DESCRIPTION
**Pull request by Kccendana**
### **I did the following in this Pull Request:**

 - Create a class Classroom with the following:
   - @label instance variable, should be initialized in the constructor.
   - Setter and getter for @label (remember about attr_accessor).
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
   - Create the has-many side (a classroom has many students).
  - Create the belongs-to side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class Book with the following:
  - @title and @author instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about attr_accessor).
- Create a class Rental with the following:
  - @date instance variable, should be initialized in the constructor.
  - Setter and getter for @date (remember about attr_accessor).
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
  - Create the has-many side of Book and Rental (a book has many rentals).
  - Create the belongs-to side of Rental and Book (a rental belongs to a book).
  - Create the has-many side of Person and Rental (a person has many rentals).
  - Create the belongs-to side of Rental and Person (a rental belongs to a person).
  - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require_relative 'book'
+require_relative 'person'
+
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(book, person)
+    @date = Date.today
```
Comment (Shedrack-Sunday) [inline rental.rb]: - [ ] Kindly set the date as an instance variable, which should be initialized in the constructor. This allows us to set a specific date, and not always "today's" date of which will not be the case everytime we login a data.

### Requirement:
image

### Sample code:
```
def initialize(date)
    @date = date
......
```

File: student.rb [hunk 2]
```diff
@@ -1,12 +1,20 @@
 require_relative 'person'
+require_relative 'classroom'
 
 class Student < Person
-  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
+  attr_accessor :classroom
+
```
Comment (Shedrack-Sunday) [inline student.rb:6]: - [Optional] Its good practice to have the accessor calls at the top of the classes and not inside. Kindly consider this 👍

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @Kccendana          👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the Ruby features ✔️
- Clean use of classcall ✔️  

# Required Changes  🌵  

- [x] The changes required are minor but very important so we avoid bugs in the application.

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by addod19 — APPROVED** [review]
## Hi @Kccendana, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Great job addressing the comments from the previous review :rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**